src/view/view.cpp

```diff
@@ -1132,6 +1145,12 @@ void wayfire_view_t::minimize_request(bool state)
     }
 }
 
+void wayfire_view_t::close_request()
+{
+    destroy_toplevel();
```

Comment:
we don't need to destroy the toolevel here. The client might not close its window. In fact, the close() function is actually just a request to the client to close.